tsdb/shard.go

```diff
@@ -178,23 +223,93 @@ func (s *Shard) WritePoints(points []Point) error {
 			}
 		}
 
-		// save the raw point data
+		// Write points to WAL bucket.
+		wal := tx.Bucket([]byte("wal"))
 		for _, p := range points {
-			bp, err := tx.CreateBucketIfNotExists(p.Key())
+			// Generate an autoincrementing index for the WAL.
+			id, _ := wal.NextSequence()
```

Comment:
The `id` that is created here, is used as a key. Data is inserted using this key. Is this data ever deleted from the WAL bucket?